File: 78-subsets/78-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = [[]]
        subset = []
        def backtrack(start):
            
            for i in range(start, len(nums)):
                subset.append(nums[i])
                ans.append(subset.copy())
                backtrack(i + 1)
                subset.pop()
        
        backtrack(0)
        return ans

File: 78-subsets/78-subsets.cpp
class Solution {
public:
    vector<int> subset;
    vector<vector<int>> ans;
    
    void generate_subsets(int start, vector<int> nums)
    {
        for(int i = start; i < nums.size(); i++)
        {
            subset.push_back(nums[i]);
            ans.push_back(subset);
            generate_subsets(i + 1, nums);
            subset.pop_back();
        }
    }
    
    vector<vector<int>> subsets(vector<int>& nums) {
        ans.push_back(subset);
        generate_subsets(0, nums);
        return ans;
    }
};